ingestion-edge/tests/integration/publish/test_healthy.py

```diff
@@ -0,0 +1,121 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, you can obtain one at http://mozilla.org/MPL/2.0/.
+
+from .helpers import IntegrationTest
+from ingestion_edge.config import METADATA_HEADERS
+from itertools import cycle, chain
+from typing import Dict, Optional
+import gzip
+import pytest
+
+# avoid haveing long strings in parametrized test
+# names by outsourcing the actual values
+DATA = {
+    b"GZIPPED": gzip.compress(b"data"),
+    b"MAX_LENGTH": b"." * (8 * 1024 * 1024),  # max data length 8MB
+    b"NON_UTF8": bytes(range(256)),
+    b"TOO_LONG": b"." * (8 * 1024 * 1024 + 1),
+    # identity values
+    b"": b"",
+    b"data": b"data",
+}
+HEADER = {
+    b"MAX_LENGTH": b"." * 1024,
+    b"NON_UTF8": bytes(set(range(256)).difference(b"\r\n"))[::-1],
+    b"TOO_LONG": b"." * 1025,
+    # identity values
+    None: None,
+    b"": b"",
+    b"header": b"header",
+}
+STRING = {
+    "MAX_LENGTH": "." * 1024,
+    "NON_UTF8": bytes(set(range(256)).difference(b"\r\n"))[::-1].decode("latin"),
+    "TOO_LONG": "." * 1025,
+    # identity values
+    "": "",
+    ".": ".",
+    "args": "args",
+    "p/a/t/h": "p/a/t/h",
+}
+
+
+@pytest.mark.parametrize(
+    "args,data,headers,uri_suffix",
+    [
+        case[:-1]
+        for case in zip(
+            cycle(["", "args", "MAX_LENGTH"]),  # args
+            cycle([b"", b"data", b"GZIPPED", b"NON_UTF8", b"MAX_LENGTH"]),  # data
+            cycle(
+                chain(
+                    [
+                        {
+                            key: value
+                            for key in METADATA_HEADERS
+                            if key != "content-length"
+                        }
+                        for value in [None, b"", b"header", b"NON_UTF8"]
+                    ],
+                    [
+                        {key: b"MAX_LENGTH"}
+                        for key in METADATA_HEADERS
+                        if key != "content-length"
+                    ],
+                )
+            ),  # headers
+            ([".", "p/a/t/h", "MAX_LENGTH"]),  # uri_suffix
+            # list of None so cycle doesn't generate infinite tests
+            [None]
+            * max(
+                5, 4 + len(set(METADATA_HEADERS.keys()).difference({"content-length"}))
+            ),
+        )
+    ],
+)
+def test_submit_success(
```

Comment:
`test_submit_success` is an expansion of `test_publish` from the former `test_api.py` file, expanded to cover more cases, especially around message and attribute size.